### PR TITLE
cmake: sync with `configure.py` (15/n)

### PR DESCRIPTION
**Pull request by tchaikov**
this is the 15th changeset of a series which tries to give an overhaul to the CMake building system. this series has two goals:
    - to enable developer to use CMake for building scylla. so they can use tools (CLion for instance) with CMake integration for better developer experience
    - to enable us to tweak the dependencies in a simpler way. a well-defined cross module / subsystem dependency is a prerequisite for building this project with the C++20 modules.

this changeset includes following changes:

 - build: cmake: add two missing tests
 - build: cmake: port more cxxflags from configure.py

### CODE REVIEW

**Comment by tchaikov** [comment]
@denesb hi Botond, could you please help review this change as well?

**Review by denesb — APPROVED** [review]

**Comment by tchaikov** [comment]
still #12629

**Comment by tchaikov** [comment]
rebased onto master HEAD

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hello scylla maintainers, could you please help queue this changeset?

**Comment by nyh** [comment]
> @scylladb/scylla-maint hello scylla maintainers, could you please help queue this changeset?

Need to wait for another CI round since you pushed a new version...

**Comment by tchaikov** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/304/

#13329

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hello scylla maintainers, could you please help queue this changeset?

**Pull request merged**
